astropy/io/fits/tests/test_core.py

```diff
@@ -1310,7 +1310,7 @@ def test_primaryhdu_zero_bytes(self):
 
         hdu_img_2880 = fits.PrimaryHDU(data=np.arange(720, dtype='i4'))
         # use safeio, a special file handler meant to fail on zero-byte writes
-        fh = safeio.CatchZeroByteWriter(open('image.fits', mode='wb'))
+        fh = safeio.CatchZeroByteWriter(open(self.temp('image.fits'), mode='wb'))
```

Comment:
@ConorMacBride - Actually could you move the commit with this change to another PR ? Otherwise backporting your PR will fail because this change has not be backported.

Reply:
It's now reverted. I'll open another PR for it.